generators/server/templates/pom.xml.ejs

```diff
@@ -690,7 +690,7 @@
             <artifactId>spring-cloud-starter-consul-config</artifactId>
         </dependency>
         <%_ } _%>
-        <%_ if (authenticationType === 'uaa') { _%>
+        <%_ if (authenticationType === 'oauth2' || authenticationType === 'uaa') { _%>
```

Comment:
Pending remark : 
(applicationType === 'gateway' && authenticationType === 'oauth2') may be safer than authenticationType === 'oauth2'